src/services/bitcoin.js

```diff
@@ -113,7 +113,7 @@ export const sendRawTransaction = async (rawTx: string): Promise<string> => {
 
 export const transactionFromPlan = (
   plan: BitcoinTransactionPlan,
-  inputSigner: (address: string) => ECPair,
+  keyPair: ECPair,
```

Comment:
Why are we doing this?

Reply:
Transactions can be made from multiple addresses, which will require different keys.

Reply:
but the wallet currently has the keyPair derived from the only mnemonic of the pillarwallet, those inputs are coming from the unspent transactions to that address for sending.

Reply:
Yes, but this service doesn't have to be aware of that limitation, IMO. We can delegate that to the caller.

```js
const { mnemonic, privateKey, path } = wallet;
let seed = privateKey;
if (mnemonic && mnemonic !== 'ENCRYPTED') {
  seed = mnemonic;
}

const root = await rootFromMnemonic(seed);

const rawTransaction = transactionFromPlan(plan, () => {
  return root.derivePath(path);
});
```

This would leave us some ground ready for multiple addresses, for example, we could have an object mapping addresses to paths, and it would be just changing it to:

```js
// const addressPathMap = { "<address>": <path> };
const rawTransaction = transactionFromPlan(plan, (address) => {
  return root.derivePath(addressPathMap[path]);
});
```

Having multiple addresses is gonna be mandatory, so I think we should have as much code ready for that as we can.  Do you think this make sense?

Reply:
Makes sense, added a ToDo and reverted the inputSigner callback 18b5299